### DISCUSSION TITLE
Incomplete contents during import file DXF to PNG/PDF

### DISCUSSION THREAD

**Discussion by thawatchai-p** [mozman/ezdxf#661] [Q&A]
Hello, 

I tried to convert files by using add-ons features of ezdxf.
After importing, it was found that there were some missing entities, for example, text boxes, and others components such as drawing layout was not included. Is it possible to fix all these issues and get the complete PNG file.

[Original Drawing File.zip](https://github.com/mozman/ezdxf/files/8370071/Original.Drawing.File.zip)

It is suppose to be like this:
![Original Drawing](https://user-images.githubusercontent.com/53156869/160584963-0a778cb3-722c-409c-ba57-65dc5cf5d5a2.jpg)

The convert objects were look like as following:
![Convert by ezdxf](https://user-images.githubusercontent.com/53156869/160581198-e7a1cf23-fddf-4e6e-87da-e16e3e5e8dff.png)

Coding was as following:

```
import ezdxf
from ezdxf import recover
from ezdxf.addons.drawing.config import Configuration, LinePolicy
from ezdxf.addons.drawing import RenderContext, Frontend
from ezdxf.addons.drawing import matplotlib
from ezdxf.addons.drawing.matplotlib import MatplotlibBackend

try:
    doc, auditor = recover.readfile('your.dxf')
except IOError:
    print(f'Not a DXF file or a generic I/O error.')
    sys.exit(1)
except ezdxf.DXFStructureError:
    print(f'Invalid or corrupted DXF file.')
    sys.exit(2)

msp = doc.modelspace()
config = Configuration.defaults()
config = config.with_changes(lineweight_scaling=0.7, line_policy=LinePolicy.ACCURATE)
dpi = 600

if not auditor.has_errors:
    matplotlib.qsave(msp, filename='drawing.png', bg='#FFFFFF', dpi=dpi, config=config)
```
Thank you.

**Comment by mozman** [comment]
If you render the modelspace you get the modelspace:

![image](https://user-images.githubusercontent.com/4230966/160631232-a63fbb8c-3a7e-416b-a406-eed8804d5ca6.png)

And the VIEWPORT entity is not supported.

**Reply by thawatchai-p** [reply]
After checking, it was found as you said. There were 1 modelspace and 2 paperspace.
```
doc = ezdxf.readfile(input_dxf_file)
doc.layouts.names_in_taborder()
```
Layout1:
<img width="765" alt="Layout1" src="https://user-images.githubusercontent.com/53156869/160664793-a7a3bd52-7be1-4c2e-be8b-9fb47f4b81da.png">
In that case, Is there any way to merge/combine those into 1 modelspace? 
Thank you again for your advice.

**Reply by mbway** [reply]
I'm not aware of anything that would fundamentally prevent adding VIEWPORT support to the drawing addon, but no-one has implemented it yet. I created the initial version of the addon and this feature wasn't a requirement for my use case.
